proxy-support/README.md

```diff
@@ -0,0 +1,273 @@
+- [To be discussed](#to-be-discussed)
+- [Current known limitations and issues:](#current-known-limitations-and-issues)
+- [Tasks, Stories and PRs](#tasks-stories-and-prs)
+  - [Tasks](#tasks)
+  - [PRs](#prs)
+<!-- /toc -->
+
+## Glossary
+
+- Management Cluster (`MC`): A cluster which hosts all ClusterAPI controllers and extend this with Giant Swarm specific features like GitOps tooling and an observability stack.
+- Workload Cluster (`WC`): A cluster created by a ClusterAPI controller, which is meant to be used by end-users.
+- Installation: An installation defines the combination of a `MC` with all the provisioned `WCs` from there.
+- App Platform: Refers to a set of features and concepts that allow to browse, install and manage the configurations of apps from the management cluster.
+- Proxy configuration: In most cases a combination of valid `http_proxy`,`https_proxy` and `no_proxy` environment variables.
+
+## Summary
+
+Goal of this proposal is to collect all requirements and information needed to create an installation from scratch behind a proxy server.
+
+## Motivation
+
+Due some security related policies, some customers are enforced to use a corporate proxy server when accessing external resources. Beside some application specific API endpoints this also affects accessing our used:
+
+- container registries (e.g. `quay.io`, `docker.io`)
+- GitOps workflow (e.g. accessing `github.com`)
+
+## Affected Components
+
+### `containerd`
+
+To make `containerd` use of a proxy configuration for pulling container images, it's required to create a `systemd` drop-in and place it under `/etc/systemd/system/containerd.service.d/http-proxy.conf`.
+
+```
+[Service]
+Environment="HTTP_PROXY={{ .HttpProxy }}"
+Environment="http_proxy={{ .HttpProxy }}"
+Environment="HTTPS_PROXY={{ .HttpsProxy }}"
+Environment="https_proxy={{ .HttpsProxy }}"
+Environment="NO_PROXY={{ .NoProxy}}"
+Environment="no_proxy={{ .NoProxy }}"
+```
+
+### workload on top of Kubernetes
+
+Every pod within a Kubernetes cluster needs a valid proxy configuration if the pod needs to access external systems.
+
+On a `MC` there are a lot of components which needs access to external systems:
+
+- `Flux` needs access to `github.com`
+- The entire monitoring and alerting stack needs access to `opsgenie`, `grafana`, ...
+- App platform components need access to an OCI artifact system (e.g. `github.com`, `azure`)
+
+As we already use `kyverno` in our `MC`, we can use the [`kyverno-policies-connectivity`] chart to inject the proxy configuration into all pods.
+
+> Please note: 
+> - due multiple valid reasons, `kyverno` doesn't mutate objects in `namespace/kube-system` per default.
+> - static pods are managed by the `kubelet`. For these pods it's not possible to use `kyverno` for mutation. Therefore it's required to inject the proxy configuration during the machine creation phase (could be achieved in `CAPI` based clusters via `kubeadm`)
+
+#### special workload
```

Comment:
I think this should be H3 header. It is parent of the ones below.

Reply:
as the objects below are H5 i think H4 is correct.